### PR TITLE
perl: disable libnsl

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
My PR https://github.com/Perl/perl5/pull/20217 to remove `libnsl` from `libswanted` in `Configure` was merged so I think we can apply this change here.  We'll be able to drop the `inreplace` once it is in a new tag.  Since we were not linking to this before, we shouldn't need a new bottle or to test dependents.

File: Formula/perl.rb [hunk 1]
```diff
@@ -30,6 +30,11 @@ class Perl < Formula
   skip_clean "lib/perl5/site_perl"
 
   def install
+    # Apply upstream commit to remove nsl from libswanted:
+    # https://github.com/Perl/perl5/commit/7e19816aa8661ce0e984742e2df11dd20dcdff18
+    # Remove with next tagged release that includes the change.
+    inreplace "Configure", " nsl ", ""
```
Comment (carlocab) [inline Formula/perl.rb:36]: Can we use a patch instead of an `inreplace`?

Comment (carlocab) [reply Formula/perl.rb:36]: Using this commit should probably work:

https://github.com/Perl/perl5/commit/7e19816aa8661ce0e984742e2df11dd20dcdff18.patch?full_index=1

Comment (danielnachun) [reply Formula/perl.rb:36]: Okay I'll try to apply that commit and switch to a patch if the commit cannot be applied cleanly.  We are at 5.34 right now and `blead` is at 5.37 so I'm not sure if the commit will apply.  Nonetheless it's definitely worth trying that first.

Comment (danielnachun) [reply Formula/perl.rb]: The commit should work based on local testing.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**